test/cpp/end2end/xds_end2end_test.cc

```diff
@@ -2240,6 +2273,8 @@ TEST_P(LdsTest, RouteMatchHasNonemptyPrefix) {
   CheckRpcSendFailure();
   EXPECT_EQ(balancers_[0]->ads_service()->lds_response_state(),
             AdsServiceImpl::NACKED);
+  EXPECT_EQ("Prefix does not start with a /",
```

Comment:
This error message does not match what should be returned for this test, which suggests that the test case is wrong.  I suggest prepending a "/" to the prefix string in the route config and seeing if we get a more accurate error message here.  If not, there's probably a bug that needs fixing.

Note that this means that this PR is already paying for itself, since it's helping us find problems (regardless of whether the problem lies in the test or in the code being tested).